packages/calcite-components/src/components/button/button.scss

```diff
@@ -1,64 +1,144 @@
+/**
+ * CSS Custom Properties
+ *
+ * These properties can be overridden using the component's tag as selector.
+ *
+ * @prop --calcite-button-background-color: Specifies the component's background color.
+ * @prop --calcite-button-border-color: Specifies the component's border color.
+ * @prop --calcite-button-corner-radius: Specifies the component's corner radius.
+ * @prop --calcite-button-corner-radius-start-start: Specifies the component's top-left corner radius. will fallback to --calcite-button-corner-radius.
+ * @prop --calcite-button-corner-radius-start-end: Specifies the component's top-right corner radius. Will fallback to --calcite-button-corner-radius.
+ * @prop --calcite-button-corner-radius-end-start: Specifies the component's bottom-left corner radius. Will fallback to --calcite-button-corner-radius.
+ * @prop --calcite-button-corner-radius-end-end: Specifies the component's bottom-right corner radius. Will fallback to --calcite-button-corner-radius.
+ * @prop --calcite-button-icon-color: Specifies the color of an icon in the component.
+ * @prop --calcite-button-loader-color: Specifies the component's loader color.
```

Comment:
Question: Do we have a specific use case for a separate loader color? This may be outdated, but the advice from the [Guidelines for Granular Icons](https://www.figma.com/design/wTABvYi73K4qaOsfvQQu4j/Guidelines-for-Granular-Icon-Colors?node-id=0-1&node-type=canvas&m=dev) specifies button loader should share the same color as icon. cc @SkyeSeitz @ashetland

Reply:
both icon-color and loader-color should inherit from text color